apps/authentication-portal/src/main/webapp/basicauth.jsp

```diff
@@ -391,23 +396,46 @@
     </div>
     <% } %>
     <%!
-        private String getRecoverAccountUrl (
-            String identityMgtEndpointContext,
-            String urlEncodedURL,
-            boolean isUsernameRecovery,
-            String urlParameters) {
-
-            return identityMgtEndpointContext + "/recoveraccountrouter.do?" + urlParameters +
-                "&isUsernameRecovery=" + isUsernameRecovery + "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
+        private String getRecoverAccountUrl(String identityMgtEndpointContext, String urlEncodedURL,
+                boolean isUsernameRecovery, String urlParameters) {
+
+            return buildUrlContext(identityMgtEndpointContext) + ACCOUNT_RECOVERY_ENDPOINT_RECOVER + "?" + urlParameters
+                    + "&isUsernameRecovery=" + isUsernameRecovery + "&callback=" + Encode
+                    .forHtmlAttribute(urlEncodedURL);
+        }
+
+        private String getRegistrationUrl(String identityMgtEndpointContext, String urlEncodedURL,
+                String urlParameters) {
+
+            return buildUrlContext(identityMgtEndpointContext) + ACCOUNT_RECOVERY_ENDPOINT_REGISTER + "?"
+                    + urlParameters + "&callback=" + Encode.forHtmlAttribute(urlEncodedURL);
```

Comment:
wondering what's the requirement to encode a query param for an html attribute. This should have been encoded for a URI component

Reply:
yes, https://github.com/OWASP/owasp-java-encoder/wiki/2)-Use-the-OWASP-Java-Encoder  section "Encode URL parameter values"

Reply:
Let's track this and proceed with the approach existed. Need to check why it had been done so